specification/cdn/resource-manager/Microsoft.Cdn/stable/2020-03-31/cdn.json

```diff
@@ -3856,6 +3862,61 @@
           }
         }
       },
+      "UrlSigningActionParameters": {
+        "description": "Defines the parameters for the Url Signing action.",
+        "required": [
+          "keyId"
+        ],
+        "properties": {
+          "keyId": {
+            "description": "Id reference of the key to be used to verify the hash and should be defined in UrlSigningKeys",
+            "type": "string"
+          },
+          "algorithm": {
+            "description": "Algorithm to use for URL signing",
+            "type": "string",
+            "enum": [
+              "SHA256"
+            ]
+          },
+          "parameterNameOverride": {
```

Comment:
I am getting confused looking at this. Is this really an override? These are paramters that define how url signing will work for that request. Please correct me if I am wrong.

Reply:
This is an override for the query parameter names. Essentially what we are trying to say is, we have three query parameters ("expires", "keyId", "signature"). We want to give the ability for the customers to change the name of these parameters so that we and the customer are on the same page.
For example, 
 `<baseurl>?expires=<timestamp>&keyid=<somekeyid>&signature=<somesignature>`
 is the default set of query parameters the customer should use. We can allow the customer to change this to something like:
 `<baseurl>?expirationtimestamp=<timestamp>&hashingkeyid=<somekeyid>&signature=<somesignature>`
if the customer chooses to do so.
We are not explicitly using the name query parameter name override here because in the future we might want to support Cookies or Headers for these values and we would have to come back and change the swagger at that time.